07_errors/alextmz/syncstore.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"sync"
+)
+
+type SyncStore struct {
+	sync.Map
+	sync.Mutex
+}
+
+func NewSyncStore() *SyncStore {
+	a := SyncStore{}
+	return &a
+}
+
+func (m *SyncStore) CreatePuppy(p *Puppy) error {
+	if _, ok := m.Load(p.ID); !ok {
+		if p.Value < 0 {
```

Comment:
This check could be done first as error should be returned regardless of puppy with given id already exists or not.

Reply:
Agree